src/ui/MiniChart/Area.svelte

```diff
@@ -10,6 +13,39 @@
   export let height: number
   export let valueKey: Props['valueKey'] = undefined
   export let style: Props['style'] = undefined
+  export let tooltipVisible = false
+  export let tooltipSyncKey = ''
+  export let formatTooltipValue = (value: number) => value.toString()
+
+  const { offset$: sharedOffset$, syncKey$, updateOffset } = getTooltipContext() ?? {}
+
+  let localOffset = 0
+
+  $: hasSharedOffset = !!sharedOffset$ && !!tooltipSyncKey && $syncKey$ === tooltipSyncKey
+  $: sharedOffset = hasSharedOffset ? $sharedOffset$ ?? 0 : 0
+  $: offset = hasSharedOffset ? sharedOffset : localOffset
+  $: currentValue = getValueAt(offset, width)
+  $: valueFormatted = currentValue !== undefined ? formatTooltipValue(currentValue) : currentValue
+
+  function getValueAt(offset: number, width: number) {
+    if (!offset) return
+    const valueIndex = Math.round((offset / width) * (data.length - 1))
+    return getValue(data[valueIndex])
+  }
+
+  function onMouseMove({ offsetX }: MouseEvent) {
+    localOffset = offsetX
+    if (tooltipSyncKey) {
+      updateOffset?.(offsetX, tooltipSyncKey)
+    }
+  }
+
+  function onMouseLeave() {
+    localOffset = 0
+    if (tooltipSyncKey) {
+      updateOffset?.(0, tooltipSyncKey)
+    }
+  }
```

Comment:
Default component should not be modified in such a way, when the goal is to support optional behavior. The more flexible and DX friendly way would be to use composition. For example, a `InteractiveArea` component, which will encapsulate tooltip logic and render the `Area` mini-chart